### PR TITLE
Add modulo function

### PR DESCRIPTION
**Pull request by GlitchlessCode**
# The Suggestion
Comes from #11 
# Implementation description
- Uses an implementation of the `fmod` algorithm
  - `x % y = x - truncate(x / y) * y`
  - The truncation was the hardest part to figure out, because unlike `floor` which always rounds down, `truncate` rounds towards 0
- Implementation is fairly efficient, and is quite precise
- Allows for both decimal and negative values in both `x` and `y`
- Catches `y=0` by failing
- Updated README to add `gm:modulo` to list of functions

### CODE REVIEW

**Comment by GlitchlessCode** [comment]
I made sure to test this as extensively as possible, and I couldn't find any edge cases where it broke (apart from `y=0`, which now correctly returns a fail). This includes int/int, int/float, float/int, float/float, positive/positive, positive/negative, negative/positive, negative/negative, extremely small values (including exponenent notation), extremely large values, and very long decimals.

File: data/gm/functions/zzz/modulo/divide.mcfunction [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#> gm:zzz/modulo/divide
+#
+# Divides var2 by var3
+#
+# ---
+# @context any
+# @within gm:modulo
+
+say divide
```
Comment (gibbsly) [inline data/gm/functions/zzz/modulo/divide.mcfunction:9]: :)

Comment (gibbsly) [reply data/gm/functions/zzz/modulo/divide.mcfunction:9]: already removed the say don't worry about it https://github.com/gibbsly/gm/commit/e6efebb2b2762a75c8f7da5bb8c75fc1d3034930

**Comment by gibbsly** [comment]
excellent work as always ty

**Comment by GlitchlessCode** [comment]
Oh woops, thanks for catching that :P
I had that for debugging purposes
